WordPress/Classes/Networking/MenusServiceRemote.h

```diff
@@ -0,0 +1,93 @@
+#import "ServiceRemoteREST.h"
+
+extern NSString * const MenusRemoteKeyID;
+extern NSString * const MenusRemoteKeyMenu;
+extern NSString * const MenusRemoteKeyMenus;
+extern NSString * const MenusRemoteKeyLocations;
+extern NSString * const MenusRemoteKeyContentID;
+extern NSString * const MenusRemoteKeyDescription;
+extern NSString * const MenusRemoteKeyLinkTarget;
+extern NSString * const MenusRemoteKeyLinkTitle;
+extern NSString * const MenusRemoteKeyName;
+extern NSString * const MenusRemoteKeyType;
+extern NSString * const MenusRemoteKeyTypeFamily;
+extern NSString * const MenusRemoteKeyTypeLabel;
+extern NSString * const MenusRemoteKeyURL;
+extern NSString * const MenusRemoteKeyItems;
+extern NSString * const MenusRemoteKeyDeleted;
+extern NSString * const MenusRemoteKeyLocationDefaultState;
+
+@class Blog;
+@class RemoteMenu;
+@class RemoteMenuItem;
+@class RemoteMenuLocation;
+
+typedef void(^MenusServiceRemoteSuccessBlock)();
+typedef void(^MenusServiceRemoteMenuRequestSuccessBlock)(RemoteMenu *menu);
+typedef void(^MenusServiceRemoteMenusRequestSuccessBlock)(NSArray<RemoteMenu *> *menus, NSArray<RemoteMenuLocation *> *locations);
+typedef void(^MenusServiceRemoteFailureBlock)(NSError *error);
+
+@interface MenusServiceRemote : ServiceRemoteREST
+
+#pragma mark - Remote queries: Creating and modifying menus
+
+/**
+ *  @brief      Create a new menu on a blog.
+ *
+ *  @param      menuName    The name of the new menu to be created.  Cannot be nil.
+ *  @param      blog        The blog to create the menu on.  Cannot be nil.
+ *  @param      success     The success handler.  Can be nil.
+ *  @param      failure     The failure handler.  Can be nil.
```

Comment:
PS: the moment you add nullability info to this method, the rest of the header file will require it too.